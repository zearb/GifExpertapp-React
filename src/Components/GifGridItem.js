import React from 'react';
import PropTypes from 'prop-types';

export default function GifGridItem({title,url}) {

    
    // console.log(props);

    return (
        <div className="card animate__animated animate__zoomIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}