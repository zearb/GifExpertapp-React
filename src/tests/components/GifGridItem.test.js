import React from 'react';
// import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GifGridItem from '../../Components/GifGridItem';

describe('Pruebas en el componente GifGridItem', () => {
    
    const title = 'DBZ';
    const url = 'urlDeUnaImagen.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    
    test('GifGridItem debe mostrarse correctamente', () => {        

        expect(wrapper).toMatchSnapshot();

    });

    test('GifGridItem debe tener la imagen con url y alt de los props', () => {

        const img = wrapper.find('img');

        // console.log(img.html());
        // console.log(img.props());
        // console.log(img.prop('src'));

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    test('GifGridItem debe tener un parrafo con el title', () => {

        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(title);

    });

    test('El div contenedor debe tener la clase animate__zoomIn',()=>{

        const div = wrapper.find('div');

        // console.log(div.prop('className'));

        expect(div.prop('className').includes('animate__zoomIn')).toBe(true);

    });
    

});
