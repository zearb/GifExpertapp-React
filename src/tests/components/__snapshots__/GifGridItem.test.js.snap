// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem GifGridItem debe mostrarse correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="DBZ"
    src="urlDeUnaImagen.jpg"
  />
  <p>
    DBZ
  </p>
</div>
`;
